lightwood/api/json_ai.py

```diff
@@ -664,50 +652,42 @@ def code_from_json_ai(json_ai: JsonAI) -> str:
     json_ai = _add_implicit_values(json_ai)
 
     # ----------------- #
-    # Instantiate encoders
-    encoder_dict = {
-        json_ai.problem_definition.target: call(
-            list(json_ai.outputs.values())[0].encoder
-        )
-    }
 
-    # Instantiate Depedencies
-    dependency_dict = {}
-    dtype_dict = {
-        json_ai.problem_definition.target: f"""'{list(json_ai.outputs.values())[0].data_dtype}'"""
-    }
+    # Instantiate data types
+    dtype_dict = json_ai.dtype_dict
 
-    # Populate features and their data-types
-    for col_name, feature in json_ai.features.items():
-        encoder_dict[col_name] = call(feature.encoder)
-        dependency_dict[col_name] = feature.dependency
-        dtype_dict[col_name] = f"""'{feature.data_dtype}'"""
+    # Populate encoders
+    encoder_dict = {}
+    for col_name, encoder in json_ai.encoders.items():
+        encoder_dict[col_name] = call(encoder)
 
     # Populate time-series specific details
     tss = json_ai.problem_definition.timeseries_settings
     if tss.is_timeseries and tss.use_previous_target:
         col_name = f"__mdb_ts_previous_{json_ai.problem_definition.target}"
-        json_ai.problem_definition.timeseries_settings.target_type = list(
-            json_ai.outputs.values()
-        )[0].data_dtype
+        target_type = json_ai.dtype_dict[json_ai.problem_definition.target]
+        json_ai.problem_definition.timeseries_settings.target_type = target_type
         encoder_dict[col_name] = call(
             lookup_encoder(
-                list(json_ai.outputs.values())[0].data_dtype,
+                target_type,
                 col_name,
                 False,
                 json_ai.problem_definition,
                 False,
                 None,
             )
         )
-        dependency_dict[col_name] = []
-        data_dtype = list(json_ai.outputs.values())[0].data_dtype
-        dtype_dict[col_name] = f"""'{data_dtype}'"""
-        json_ai.features[col_name] = Feature(encoder=encoder_dict[col_name], data_dtype=data_dtype)
+
+        dtype_dict[col_name] = f"""'{target_type}'"""
+        # @TODO: Is populating the json_ai at this stage even necessary?
```

Comment:
For time series it is, as we automatically generate encoders for e.g. previous target columns.